src/librustc_mir/build/matches/mod.rs

```diff
@@ -377,7 +385,8 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                               span: Span,
                               arm_blocks: &mut ArmBlocks,
                               mut candidates: Vec<Candidate<'pat, 'tcx>>,
-                              mut block: BasicBlock)
+                              mut block: BasicBlock,
+                              opt_imaginary_target_block: &mut Option<BasicBlock>)
```

Comment:
You should add a comment somewhere explaining the purpose of the `opt_imaginary_target_block` to chain things together